diff --git a/link.d.ts b/link.d.ts
index 138998c63e593d77df39d4af1ef26d0ed29cd625..23459aa6b8e722bccbaf6f060e89926f5f503cf3 100644
--- a/link.d.ts
+++ b/link.d.ts
@@ -3,11 +3,18 @@ import type { UnpackProps } from "./tools/UnpackProps";
 type HTMLAnchorProps = DetailedHTMLProps<AnchorHTMLAttributes<HTMLAnchorElement>, HTMLAnchorElement>;
 export interface RegisterLink {
 }
-export type RegisteredLinkProps = RegisterLink extends {
-    Link: infer Link;
-} ? Omit<UnpackProps<Link>, "children"> | (Omit<HTMLAnchorProps, "children" | "href"> & {
-    href: string;
-}) : Omit<HTMLAnchorProps, "children">;
+export type RegisteredLinkProps = RegisterLink extends { Link: infer Link }
+    ? LinkPropsWithoutChildren<UnpackProps<Link>>
+    : RegisterLink extends { LinkProps: infer LinkProps }
+    ? LinkPropsWithoutChildren<LinkProps>
+    : Omit<HTMLAnchorProps, "children">;
+
+type LinkPropsWithoutChildren<LinkProps> =
+    | Omit<LinkProps, "children">
+    | (Omit<HTMLAnchorProps, "children" | "href"> & {
+          href: string;
+      });
+
 declare let Link: React.ComponentType<RegisteredLinkProps & {
     children: ReactNode;
 }>;
