diff --git a/dist/esm/parse.js b/dist/esm/parse.js
index a3c64463f5eba87ca5d51b71f320bcda313eefa2..27ffe750447811a7e54ae376f28107e671d715cc 100644
--- a/dist/esm/parse.js
+++ b/dist/esm/parse.js
@@ -1,5 +1,5 @@
 import { NodeType, parse, } from 'node-html-parser';
-import * as cssTree from 'css-tree';
+import cssTree from 'css-tree';
 const { generate, parse: cssParse } = cssTree;
 import supportedStyles from './supportedStyles.js';
 import camelize from './camelize.js';
